### PR TITLE
Add JvmOverloads for java api

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Add @JvmOverloads where applicable to provide better experience for Java users.</changelog>`.

### Summary of changes

- Add `@JvmOverloads` to constructors and methods that have default Kotlin parameters
- Add overloaded methods for `MapboxMap#loadStyleUri`, `MapboxMap#loadStyleJson`, `MapboxMap#loadStyle`
- Add test code to check Java API usability

### CODE REVIEW

**Review by axti — APPROVED** [review]

**Review by axti — APPROVED** [review]

File: android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/CarJavaInterfaceTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import static com.mapbox.maps.extension.androidauto.MapboxCarUtilsKt.initMapSurface;
+
+import androidx.car.app.Session;
+
+import com.mapbox.maps.MapInitOptions;
+import com.mapbox.maps.extension.androidauto.MapSurfaceReadyCallback;
+import com.mapbox.maps.extension.androidauto.OnMapScaleListener;
+import com.mapbox.maps.extension.androidauto.OnMapScrollListener;
+
+public class CarJavaInterfaceTest {
```
Comment (alexander-kulikovskii) [inline android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/CarJavaInterfaceTest.java:12]: I'm not sure that it's a test. More like validator or checker

**Comment by kiryldz** [comment]
@mapbox/maps-android how can we make sure we did not miss anything as part of this PR? any ideas?

**Comment by axti** [comment]
Maybe we can use list of possible constructors and call them in a loop?
```
Class classSomeClass = SomeClass.class;
Constructor[] constructors = classSomeClass.getConstructors();
```

File: android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/CarJavaInterfaceTest.java (hunk 1 above)
Comment (Chaoba) [reply android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/CarJavaInterfaceTest.java:12]: Yes, just verify the Java API

Comment (alexander-kulikovskii) [reply android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/CarJavaInterfaceTest.java:12]: so maybe rename it?

**Comment by Chaoba** [comment]
> @mapbox/maps-android how can we make sure we did not miss anything as part of this PR? any ideas?

We can't 100% make sure this pr covers all the places, but we can keep it in mind and continue iterating on this point if encounter the method need add `JvmOverloads` annotation in the future..


> Maybe we can use list of possible constructors and call them in a loop?

Not only constructors, other APIs that have default param need add this annotation too.

File: android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/CarJavaInterfaceTest.java (hunk 1 above)
Comment (Chaoba) [reply android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/CarJavaInterfaceTest.java]: Renamed to xxxChecker

**Review by alexander-kulikovskii — APPROVED** [review]

**Comment by pengdev** [comment]
> Maybe we can use list of possible constructors and call them in a loop?
> 
> ```
> Class classSomeClass = SomeClass.class;
> Constructor[] constructors = classSomeClass.getConstructors();
> ```

I think we do have an API checker, will it print these info? cc @Chaoba

**Comment by Chaoba** [comment]
> I think we do have an API checker, will it print these info? cc @Chaoba

That's for checking API change, doesn't have such infos.

**Pull request merged**